### ISSUE TITLE
For array field, parser performs a partial update instead of full replacement

### ISSUE THREAD

**Issue by y2mao** [BurntSushi/toml#99]
Following code:

```
type Config struct {
	Hosts []string
}

func main() {
	cfg := Config{
		Hosts: []string{"aaa", "bbb"},
	}

	t := `Hosts = ["ccc"]`
	if _, err := toml.Decode(t, &cfg); err != nil {
		log.Fatal(err)
	}

	fmt.Printf("%v", cfg.Hosts)
}
```

Expect Result: `[ccc]`
Actual Result:  `[ccc, bbb]`

It seems parser performed a partial update instead of a full replacement. Is that possible to change parser's behaviour to full replacement?  Thanks!

**Comment by y2mao** [comment]
Alright, It seems a bug. I just found the open issue here #49

**Comment by cespare** [comment]
Fixed by #84.
